djoser/utils.py

```diff
@@ -62,48 +44,74 @@ def post(self, request):
+        self.user = user
+        self.domain = domain
+        self.site_name = site_name
+        self.protocol = protocol
+
+    @classmethod
+    def from_request(cls, request, user=None, from_email=None):
+        site = get_current_site(request)
+        return cls(
+            from_email=getattr(django_settings, 'DEFAULT_FROM_EMAIL', from_email),
+            user=user or request.user,
+            domain=django_settings.DJOSER.get('DOMAIN') or site.domain,
+            site_name=django_settings.DJOSER.get('SITE_NAME') or site.name,
+            protocol='https' if request.is_secure() else 'http',
+        )
+
+    def create(self,):
+        assert self.plain_body_template_name or self.html_body_template_name
+        context = self.get_context()
+        subject = loader.render_to_string(self.subject_template_name, context)
+        subject = ''.join(subject.splitlines())
+
+        if self.plain_body_template_name:
+            plain_body = loader.render_to_string(self.plain_body_template_name, context)
+            email_message = EmailMultiAlternatives(subject, plain_body, self.from_email, [self.user.email])
+            if self.html_body_template_name:
+                html_body = loader.render_to_string(self.html_body_template_name, context)
+                email_message.attach_alternative(html_body, 'text/html')
+        else:
+            html_body = loader.render_to_string(self.html_body_template_name, context)
+            email_message = EmailMessage(subject, html_body, self.from_email, [self.user.email])
+            email_message.content_subtype = 'html'
+        return email_message
 
-    def get_send_email_kwargs(self, user):
+    def get_context(self):
         return {
-            'from_email': getattr(django_settings, 'DEFAULT_FROM_EMAIL', None),
-            'to_email': user.email,
-            'context': self.get_email_context(user),
+            'user': self.user,
+            'domain': self.domain,
+            'site_name': self.site_name,
+            'uid': encode_uid(self.user.pk),
+            'token': self.token_generator.make_token(self.user),
+            'protocol': self.protocol,
         }
 
-    def get_send_email_extras(self):
-        return {
-            'subject_template_name': self.get_subject_template_name(),
-            'plain_body_template_name': self.get_plain_body_template_name(),
-            'html_body_template_name': self.get_html_body_template_name(),
-        }
 
-    def get_subject_template_name(self):
-        return self.subject_template_name
+class UserActivationEmailFactory(UserEmailFactoryBase):
+    subject_template_name = 'activation_email_subject.txt'
+    plain_body_template_name = 'activation_email_body.txt'
 
-    def get_plain_body_template_name(self):
-        return self.plain_body_template_name
+    def get_context(self):
+        context = super(UserActivationEmailFactory, self).get_context()
+        context['url'] = settings.get('ACTIVATION_URL').format(**context)
```

Comment:
why don't you use django_settings ? You should don't use your own settings : 


> Also note that your code should not import from either global_settings or your own settings file. django.conf.settings abstracts the concepts of default settings and site-specific settings; it presents a single interface. It also decouples the code that uses settings from the location of your settings.

https://docs.djangoproject.com/en/dev/topics/settings/#using-settings-in-python-code